### PR TITLE
fix(#895): detect scope callbacks with this.cb and global.Scope.cb patterns

### PR DESCRIPTION
**Pull request by jlaustill**
## Summary

- Fixes **Bug A** from #895: Scope callbacks using `this.cb` or `global.ScopeName.cb` patterns now generate correct pointer-based signatures
- Extends callback detection to scan scope member functions, not just standalone functions
- Resolves `this.member` → `CurrentScope_member` and `global.Scope.member` → `Scope_member`

## Background

PR #897 fixed callback signature generation for global functions, but scope callbacks remained broken because the analyzer didn't handle these reference patterns.

## Test plan

- [x] 47/47 FunctionCallAnalyzer unit tests pass (including 2 new tests)
- [x] 5414/5414 unit tests pass  
- [x] 972/972 integration tests pass
- [x] All quality checks pass (prettier, oxlint, spelling, dead code, TypeScript)

## Remaining work for #895

- **Bug B**: Opaque type local variables (`widget_t w = widget_create()` should be `widget_t* w`) - requires detecting forward-declared types
- **Bug C**: Header/implementation signature mismatch - requires header generator changes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by jlaustill — COMMENTED** [review]
## Review Summary

**PR:** #903 — fix(#895): detect scope callbacks with this.cb and global.Scope.cb patterns
**Reviewed by:** Claude (automated review)

### Overview
This PR correctly extends callback detection to scope member functions by tracking the current scope during scanning and resolving `this.member` and `global.Scope.member` patterns. The implementation is clean and well-tested.

### Findings

| Severity | Count |
|----------|-------|
| Critical | 0 |
| Important | 1 |
| Moderate | 2 |
| Nit | 0 |

### Key Issues

1. **[Important]** Instance state `scanCurrentScope` adds class state when the project prefers stateless patterns - see inline comment
2. **[Moderate]** Test duplication - both tests have nearly identical setup code that could be extracted
3. **[Moderate]** Edge case: `this.cb` outside a scope context silently falls through rather than being explicitly handled

### What Looks Good

- Clear issue comments linking back to #895
- Good separation of pattern matching (three distinct regex patterns with clear comments)
- Comprehensive test coverage for both `this.` and `global.Scope.` patterns
- The `continue` after processing `functionDeclaration` correctly avoids checking for `scopeDeclaration` on the same node

---
🤖 *Automated review by Claude Code*

File: src/transpiler/logic/analysis/__tests__/FunctionCallAnalyzer.test.ts [hunk 1]
```diff
@@ -1024,5 +1024,115 @@ describe("FunctionCallAnalyzer", () => {
         true,
       );
     });
+
+    it("should detect scope callback with this.member pattern (Issue #895 Bug A)", () => {
```
[the remark this answers is not in the document]
Comment (jlaustill) [reply src/transpiler/logic/analysis/__tests__/FunctionCallAnalyzer.test.ts]: Fixed. Extracted addWidgetCallbackSymbols() helper function that both tests now use. This reduces the duplication and makes maintenance easier.
